tm2py/core/emme.py

```diff
@@ -0,0 +1,475 @@
+"""Emme related shared code and core components.
+
+"""
+
+
+from contextlib import contextmanager as _context
+import multiprocessing as _multiprocessing
+import re as _re
+from socket import error as _socket_error
+from typing import List, Union, Any, Dict
+
+import numpy as _np
+import openmatrix as _omx
+
+# PyLint cannot build AST from compiled Emme libraries
+# so disabling relevant import module checks
+# pylint: disable=E0611, E0401, E1101
+from inro.emme.database.scenario import Scenario as EmmeScenario
+from inro.emme.database.matrix import Matrix as EmmeMatrix
+import inro.emme.desktop.app as _app
+import inro.modeller as _m
+
+NumpyArray = _np.array
+EmmeDesktopApp = _app.App
+EmmeModeller = _m.Modeller
+
+# Cache running Emme projects from this process (simple singleton implementation)
```

Comment:
docstrings for parameters go under them. See: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html